test/grpc/route_guide/route-chat.cc

```diff
@@ -0,0 +1,80 @@
+#include <iostream>
+#include <string>
+
+#include "eventuals/do-all.h"
+#include "eventuals/event-loop.h"
+#include "eventuals/finally.h"
+#include "eventuals/foreach.h"
+#include "eventuals/head.h"
+#include "eventuals/let.h"
+#include "eventuals/promisify.h"
+#include "eventuals/then.h"
+#include "gtest/gtest.h"
+#include "test/grpc/route_guide/route-guide-test.h"
+
+using grpc::Status;
+
+using eventuals::DoAll;
+using eventuals::Finally;
+using eventuals::Iterate;
+using eventuals::operator*;
+using eventuals::Foreach;
+using eventuals::Head;
+using eventuals::Let;
+using eventuals::Then;
+using eventuals::grpc::Stream;
+
+using routeguide::RouteNote;
+
+TEST_F(RouteGuideTest, RouteChatTest) {
+  const std::vector<std::tuple<std::string, long, long>> data = {
+      {"First message", 0, 0},
+      {"Second message", 0, 1},
+      {"Third message", 1, 0},
+      {"Fourth message", 0, 0}};
```

Comment:
You're using a tuple where you should just use a "struct", or in this case, a protobuf:
```suggestion
  const std::vector<RouteNote> notes = {
      MakeRouteNote("First message", 0, 0),
      MakeRouteNote("Second message", 0, 1),
      MakeRouteNote("Third message", 1, 0),
      MakeRouteNote("Fourth message", 0, 0)};
```

Does this also mean that we can delete the version of `MakeRouteNote` that takes a tuple`?